ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/editor/plugin/GeneralInfoSection.java

```diff
@@ -400,4 +451,49 @@ protected void createSingleton(Composite parent, FormToolkit toolkit, IActionBar
 		}));
 	}
 
+	protected void createBundleShape(Composite parent, FormToolkit toolkit, IActionBars actionBars, String label) {
+		Label l = toolkit.createLabel(parent, label, SWT.BORDER);
+		TableWrapData td = new TableWrapData();
+		td.colspan = 3;
+		l.setLayoutData(td);
+		Composite c = new Composite(parent, SWT.NONE);
+		GridLayout gd = new GridLayout(3, true);
+		gd.marginLeft = 0;
+		gd.marginRight = 0;
+		gd.horizontalSpacing = 2;
+		c.setLayout(gd);
+		fBundleShapeJar = toolkit.createButton(c, BUNDLESHAPE_JAR, SWT.RADIO);
+		fBundleShapeDir = toolkit.createButton(c, BUNDLESHAPE_DIR, SWT.RADIO);
+		fBundleShapeDefault = toolkit.createButton(c, "default", SWT.RADIO); //$NON-NLS-1$
+		fBundleShapeJar.setEnabled(true);
+		fBundleShapeDir.setEnabled(true);
+		fBundleShapeDefault.setEnabled(true);
+		fBundleShapeDefault.setSelection(true);
+		fBundleShapeJar.addListener(SWT.Selection, event -> {
+			if (fBundleShapeJar.getSelection()) {
+				createBundleShapeHeader(BUNDLESHAPE_JAR);
+			}
+		});
+		fBundleShapeDir.addListener(SWT.Selection, event -> {
+			if (fBundleShapeDir.getSelection()) {
+				createBundleShapeHeader(BUNDLESHAPE_DIR);
+			}
+		});
+		fBundleShapeDefault.addListener(SWT.Selection, event -> {
+			if (fBundleShapeDefault.getSelection()) {
+				if (getBundle() instanceof Bundle bundle) {
```

Comment:
It looks like the cast to `Bundle` is not necessary `IBundle` provides all the called methods.

Reply:
I agree.